### DISCUSSION TITLE
Published stream connection keeps connecting and disconnecting

### DISCUSSION THREAD

**Discussion by KAMTRIC** [bluenviron/mediamtx#1728] [Questions]
I'm publishing an RTSP video stream from a Skydio X2E controller to MediaMTX v0.21.4, running on Rocky 8 Linux OS to make it available to mobile clients. When the stream is published it connects to the server on port 8554. Then after some exchanges between the server and the controller the server then responds with a 400 bad request and shuts down the connection with the following "[RTSP] [conn 192.168.13.57:56800] closed (connection is linked to another session)". The controller then continues to connect and then subsequently shutdown, repeating over and over until the server is shut down. Below is one iteration from the log file (full log file attached). I used another video streamer to test the server and it works fine.

Has anyone seen this before? I'm looking for any possible settings I may have wrong in the MediaMTX server or issues coming from the drone controller stream that may cause this. 

2023/04/18 16:47:58 INF rtsp-simple-server v0.21.4
2023/04/18 16:47:58 DEB [path cam2] created
2023/04/18 16:47:58 DEB path manager created
2023/04/18 16:47:58 INF [RTSP] listener opened on :8554 (TCP)
2023/04/18 16:48:24 INF [RTSP] [conn 192.168.13.57:56800] opened
2023/04/18 16:48:24 DEB [RTSP] [conn 192.168.13.57:56800] [c->s] OPTIONS rtsp://192.168.13.50:8554/cam2 RTSP/1.0
CSeq: 1
User-Agent: com.pedro.rtsp 2.1.7

2023/04/18 16:48:24 DEB [RTSP] [conn 192.168.13.57:56800] [s->c] RTSP/1.0 200 OK
CSeq: 1
Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
Server: gortsplib

2023/04/18 16:48:24 DEB [RTSP] [conn 192.168.13.57:56800] [c->s] ANNOUNCE rtsp://192.168.13.50:8554/cam2 RTSP/1.0
CSeq: 2
Content-Length: 446
Content-Type: application/sdp
User-Agent: com.pedro.rtsp 2.1.7

v=0
o=- 0 0 IN IP4 127.0.0.1
s=Unnamed
i=N/A
c=IN IP4 192.168.13.50
t=0 0
a=recvonly
m=video 0 RTP/AVP 96
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1; sprop-parameter-sets=Z2QAH6y0AoAt0rAQEBQD+UCs7msoADaFCag=,aO4G4sA=
a=control:streamid=0
m=audio 0 RTP/AVP 97
a=rtpmap:97 MPEG4-GENERIC/32000/2
a=fmtp:97 profile-level-id=1; mode=AAC-hbr; config=1290; sizelength=13; indexlength=3; indexdeltalength=3
a=control:streamid=1

2023/04/18 16:48:24 INF [RTSP] [session 5d278d69] created by 192.168.13.57:56800
2023/04/18 16:48:24 DEB [RTSP] [conn 192.168.13.57:56800] [s->c] RTSP/1.0 200 OK
CSeq: 2
Server: gortsplib

2023/04/18 16:48:24 DEB [RTSP] [conn 192.168.13.57:56800] [c->s] SETUP rtsp://192.168.13.50:8554/cam2/streamid=0 RTSP/1.0
CSeq: 3
Transport: RTP/AVP/TCP;unicast;interleaved=0-1;mode=record
User-Agent: com.pedro.rtsp 2.1.7

2023/04/18 16:48:24 DEB [RTSP] [conn 192.168.13.57:56800] [s->c] RTSP/1.0 200 OK
CSeq: 3
Server: gortsplib
Session: e01a1a28-686c-4adb-ac1b-e21f77d1129c
Transport: RTP/AVP/TCP;unicast;interleaved=0-1

2023/04/18 16:48:24 DEB [RTSP] [conn 192.168.13.57:56800] [c->s] SETUP rtsp://192.168.13.50:8554/cam2/streamid=1 RTSP/1.0
CSeq: 4
Session: e01a1a28
Transport: RTP/AVP/TCP;unicast;interleaved=2-3;mode=record
User-Agent: com.pedro.rtsp 2.1.7

2023/04/18 16:48:24 DEB [RTSP] [conn 192.168.13.57:56800] [s->c] RTSP/1.0 400 Bad Request
CSeq: 4
Server: gortsplib

2023/04/18 16:48:24 INF [RTSP] [conn 192.168.13.57:56800] closed (connection is linked to another session)
2023/04/18 16:48:24 INF [RTSP] [session 5d278d69] destroyed (not in use)
2023/04/18 16:48:25 INF [RTSP] [conn 192.168.13.57:56802] opened
2023/04/18 16:48:25 DEB [RTSP] [conn 192.168.13.57:56802] [c->s] OPTIONS rtsp://192.168.13.50:8554/cam2 RTSP/1.0
CSeq: 1
User-Agent: com.pedro.rtsp 2.1.7

Full Log File:
[rtsp-simple-server-18Apr2023.log](https://github.com/aler9/mediamtx/files/11277047/rtsp-simple-server-18Apr2023.log)

**Comment by aler9** [comment]
Hello, the reason is that client `com.pedro.rtsp` doesn't like dashes in the session header. The server sends to the client this session ID:

```
Session: e01a1a28-686c-4adb-ac1b-e21f77d1129c
```

And the client transforms it into

```
Session: e01a1a28
```

resulting in a connection error.

Since this is not the first time someone reports this issue, the server will be changed in order to serve session IDs without dashes.

**Comment by aler9** [comment]
fixed by https://github.com/bluenviron/gortsplib/pull/278

**Comment by KAMTRIC** [comment]
Alessandro,

Thank you for taking the time to dig into the problem. I'm a complete noob
with all this, so could you tell which version of MediaMTX the fix will be
incorporated in? Thanks.

Ken

**Reply by aler9** [reply]
In the one that will be released in the next days.

**Comment by KAMTRIC** [comment]
Thank you.

[a comment by an automated account is not shown]
